darts/models/forecasting/regression_model.py

```diff
@@ -324,30 +324,129 @@ def _create_lagged_data(
         lags_past_covariates = self.lags.get("past")
         lags_future_covariates = self.lags.get("future")
 
-        training_samples, training_labels, _ = _create_lagged_data(
-            target_series=target_series,
-            output_chunk_length=self.output_chunk_length,
-            past_covariates=past_covariates,
-            future_covariates=future_covariates,
-            lags=lags,
-            lags_past_covariates=lags_past_covariates,
-            lags_future_covariates=lags_future_covariates,
-            max_samples_per_ts=max_samples_per_ts,
-            multi_models=self.multi_models,
-        )
+        # ensure list of TimeSeries format
+        if isinstance(target_series, TimeSeries):
+            target_series = [target_series]
+            past_covariates = [past_covariates] if past_covariates else None
+            future_covariates = [future_covariates] if future_covariates else None
+
+        training_samples, training_labels = [], []
+        for i, target in enumerate(target_series):
+            samples_i, labels_i, _ = create_lagged_training_data(
+                target_series=target,
+                output_chunk_length=self.output_chunk_length,
+                past_covariates=past_covariates[i] if past_covariates else None,
+                future_covariates=future_covariates[i] if future_covariates else None,
+                lags=lags,
+                lags_past_covariates=lags_past_covariates,
+                lags_future_covariates=lags_future_covariates,
+                max_samples_per_ts=max_samples_per_ts,
+                multi_models=self.multi_models,
+                check_inputs=False,
+            )
+            # Remove redundant sample axis:
+            samples_i = samples_i[:, :, 0]
+            labels_i = labels_i[:, :, 0]
+            training_samples.append(samples_i)
+            training_labels.append(labels_i)
 
-        training_samples = _add_static_covariates(
-            self,
+        training_samples = self._add_static_covariates(
             training_samples,
             target_series,
-            *self.extreme_lags,
-            past_covariates=past_covariates,
-            future_covariates=future_covariates,
-            max_samples_per_ts=max_samples_per_ts,
         )
```

Comment:
wouldn't calling _add_static_covariates() better done inside the preceding for loop? in my opinion, this would help simplify the _add_static_covariates() internal logic since, if it only takes a specific target (from the input sequence) and its specific features, than one can avoid looping twice over the series inside the _add_static_covariates(), if I understood the implementation correctly. WDYT?
In the current _add_static_covariates() my assumption was that the function will receive all the features and the function should compute back everything it needs in terms of length and width of features, since I was expecting a change in the _create_lagged_data() outputs, but this might not be potentially relevant anymore with the changes you made.

Reply:
It would still be necessary though to go through all the series in the input sequence once, prior, to collect the static covariates information from all of them.

Reply:
Definitely - in my opinion, the static covariates would ideally be added inside of `create_lagged_data` after each 'block' has been formed, but that would probably be a bit clumsy to implement at the moment since the process of computing the static covariates requires the `n_features_in_` attribute of the `RegressionModel` object. Perhaps something to think about for a future PR?